Assets/MRTK/SDK/Features/UX/Scripts/Pointers/BaseControllerPointer.cs

```diff
@@ -273,7 +273,13 @@ public string PointerName
         }
 
         /// <inheritdoc />
-        public IMixedRealityInputSource InputSourceParent { get; protected set; }
+        public IMixedRealityInputSource InputSourceParent
+        {
+            get { return base.Controller?.InputSource; }
```

Comment:
This will still potentially return `null` though, right? Or are we able to avoid some sort of race condition that the original bug was describing?

Reply:
It can still return null, but this makes it so that we're always referring to the controller's input source, rather than having a stale cached reference. In the base class, they don't cache the Input source, which I think is why this bug only occurs here.

https://github.com/microsoft/MixedRealityToolkit-Unity/blob/f348585c5430b6118e57d1afd8275bf9177795af/Assets/MRTK/SDK/Features/Input/Handlers/ControllerPoseSynchronizer.cs#L137-L138